src/Components/Server/src/Builder/RazorComponentsEndpointRouteBuilderExtensions.cs

```diff
@@ -0,0 +1,193 @@
+        /// <returns>The <see cref="RazorComponentsEndpointBuilder"/>.</returns>
+        public static RazorComponentsEndpointBuilder MapComponentsHub<TComponent>(
+            this IEndpointRouteBuilder routes,
+            string path,
+            string selector)
+        {
+            if (routes == null)
+            {
+                throw new ArgumentNullException(nameof(routes));
+            }
+
+            if (path == null)
+            {
+                throw new ArgumentNullException(nameof(path));
+            }
+
+            if (selector == null)
+            {
+                throw new ArgumentNullException(nameof(selector));
+            }
+
+            return routes.MapComponentsHub(path, typeof(TComponent), selector);
+        }
+
+        /// <summary>
+        /// Maps the SignalR Components Hub to the path <see cref="ComponentsHub.DefaultPath"/> and associates
+        /// the component <paramref name="componentType"/> to this hub instance.
+        /// The DOM selector for <paramref name="componentType"/> is the type name in lowercase.
+        /// </summary>
+        /// <param name="routes">The <see cref="RouteBuilder"/>.</param>
+        /// <param name="componentType">The first <see cref="IComponent"/> associated with this <see cref="ComponentsHub"/>.</param>
+        /// <returns>The <see cref="RazorComponentsEndpointBuilder"/>.</returns>
+        public static RazorComponentsEndpointBuilder MapComponentsHub(
+            this IEndpointRouteBuilder routes,
+            Type componentType)
+        {
+            if (routes == null)
+            {
+                throw new ArgumentNullException(nameof(routes));
+            }
+
+            if (componentType == null)
+            {
+                throw new ArgumentNullException(nameof(componentType));
+            }
+
+            return routes.MapComponentsHub(ComponentsHub.DefaultPath, componentType);
+        }
+
+        /// <summary>
+        /// Maps the SignalR Components Hub to the path <paramref name="path"/> and associates
+        /// the component <paramref name="componentType"/> to this hub instance. The DOM selector for <paramref name="componentType"/>
+        /// is the type name in lowercase.
+        /// The DOM selector for <paramref name="componentType"/> is the type name in lowercase.
+        /// </summary>
+        /// <param name="routes">The <see cref="RouteBuilder"/>.</param>
+        /// <param name="path">The path to map to which the <see cref="ComponentsHub"/> will be mapped.</param>
+        /// <param name="componentType">The first <see cref="IComponent"/> associated with this <see cref="ComponentsHub"/>.</param>
+        /// <returns>The <see cref="RazorComponentsEndpointBuilder"/>.</returns>
+        public static RazorComponentsEndpointBuilder MapComponentsHub(
+            this IEndpointRouteBuilder routes,
+            string path,
+            Type componentType)
```

Comment:
Should `componentType` be first in each of these ones that take a type?

Reply:
I wanted to preserve the order between generic/non-generic and have path be the first parameter, followed by the type and the selector. There is also the fact that you can't have both `MapComponent<TComponent>(string path) and MapComponent<TComponent>(string selector)` overloads and I chose to expose `MapComponent<TComponent>(string path) because I feel configuring the path is more prominent`

Reply:
I'm killing the overloads without the selector, but I feel that the path should come first as you are configuring a route at the end of the day and other Map* overloads do the same thing.

Reply:
I actually ended up moving the path last. As I expect <TCompo>("app") to be the most used overload.